From f87b4919c4303ef93af324fe71e0dd0310ba8ff1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Fri, 22 Jul 2016 11:11:44 +0200
Subject: [PATCH 09/14] contrib/ffmpeg: use audiotoolbox on iOS for AC-3

This enforces the use of the audiotoolbox decoders for AC-3 and E-AC-3
as introduced in iOS 9 and disables the use of the MLP decoder to be in
compliance with the App Store ToS.
---
 .../avcodec-enable-audiotoolbox-ac3.patch     | 20 +++++++++++++++++++
 contrib/src/ffmpeg/rules.mak                  |  9 +++++++++
 2 files changed, 29 insertions(+)
 create mode 100644 contrib/src/ffmpeg/avcodec-enable-audiotoolbox-ac3.patch

diff --git a/contrib/src/ffmpeg/avcodec-enable-audiotoolbox-ac3.patch b/contrib/src/ffmpeg/avcodec-enable-audiotoolbox-ac3.patch
new file mode 100644
index 0000000000..2d60cf7140
--- /dev/null
+++ b/contrib/src/ffmpeg/avcodec-enable-audiotoolbox-ac3.patch
@@ -0,0 +1,20 @@
+Only in ffmpeg: .DS_Store
+diff -ru ffmpeg/libavcodec/audiotoolboxdec.c ffmpeg-fix/libavcodec/audiotoolboxdec.c
+--- ffmpeg/libavcodec/audiotoolboxdec.c	2016-05-09 11:44:29.000000000 +0200
++++ ffmpeg-fix/libavcodec/audiotoolboxdec.c	2016-05-09 14:01:04.000000000 +0200
+@@ -345,7 +345,6 @@
+         avctx->bit_rate = bit_rate;
+         in_format.mSampleRate = avctx->sample_rate;
+ #endif
+-#if CONFIG_AC3_AT_DECODER || CONFIG_EAC3_AT_DECODER
+     } else if (pkt && pkt->size >= 7 &&
+                (avctx->codec_id == AV_CODEC_ID_AC3 ||
+                 avctx->codec_id == AV_CODEC_ID_EAC3)) {
+@@ -358,7 +357,6 @@
+         in_format.mChannelsPerFrame = hdr.channels;
+         avctx->frame_size = hdr.num_blocks * 256;
+         avctx->bit_rate = hdr.bit_rate;
+-#endif
+     } else {
+         in_format.mSampleRate = avctx->sample_rate ? avctx->sample_rate : 44100;
+         in_format.mChannelsPerFrame = avctx->channels ? avctx->channels : 1;
diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 12fb917d56..26fa669173 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -39,6 +39,12 @@ FFMPEGCONF = \
 ifdef HAVE_DARWIN_OS
 FFMPEGCONF += \
 	--disable-securetransport
+ifdef HAVE_IOS
+FFMPEGCONF += \
+	--disable-decoder=mlp \
+	--disable-demuxer=mlp \
+	--disable-parser=mlp
+endif
 endif
 
 DEPS_ffmpeg = zlib $(DEPS_zlib) gsm $(DEPS_gsm) openjpeg $(DEPS_openjpeg)
@@ -230,6 +236,9 @@ ffmpeg: ffmpeg-$(FFMPEG_VERSION).tar.xz .sum-ffmpeg
 	$(APPLY) $(SRC)/ffmpeg/0001-ffmpeg-add-target_os-support-for-emscripten.patch
 	$(APPLY) $(SRC)/ffmpeg/0001-dxva2_hevc-don-t-use-frames-as-reference-if-they-are.patch
 	$(APPLY) $(SRC)/ffmpeg/0001-lavc-aarch64-fix-relocation-out-of-range-error.patch
+ifdef HAVE_IOS
+	$(APPLY) $(SRC)/ffmpeg/avcodec-enable-audiotoolbox-ac3.patch
+endif
 	$(MOVE)
 
 .ffmpeg: ffmpeg
-- 
2.32.1 (Apple Git-133)

